src/components/datetime.rs

```diff
@@ -543,4 +598,61 @@ mod tests {
             }
         );
     }
+
+    fn create_diff_setting(
+        smallest: TemporalUnit,
+        increment: u32,
+        rounding_mode: TemporalRoundingMode,
+    ) -> DifferenceSettings {
+        DifferenceSettings {
+            largest_unit: None,
+            smallest_unit: Some(smallest),
+            increment: Some(unsafe { RoundingIncrement::new_unchecked(increment) }),
```

Comment:
I would prefer if we didn't use `unsafe` here. We can just use `try_new` and unwrap.